routes/error-tracker.js

```diff
@@ -128,88 +85,60 @@ function handler(req, res, params) {
       return null;
     }
 
-    const stack = standardizeStackTrace(
-      safeDecodeURIComponent(params.s || ''),
-      message
-    );
+    const stack = standardizeStackTrace(stacktrace, message);
     if (ignoreMessageOrException(message, stack)) {
       res.sendStatus(statusCodes.BAD_REQUEST);
       return null;
     }
-
-    const normalizedMessage = /^[A-Z][a-z]+: /.test(message)
-      ? message
-      : `Error: ${message}`;
-    const event = {
-      serviceContext: {
-        service: errorType,
-        version: version,
-      },
-      message: normalizedMessage,
-      context: {
-        httpRequest: {
-          method: req.method,
-          url: req.originalUrl,
-          userAgent: req.get('User-Agent'),
-          referrer: referrer,
-        },
-      },
-    };
-
-    if (req.method === 'POST') {
-      event.context.httpRequest.url += '?' + querystring.stringify(params);
-    }
-
-    const metaData = {
-      labels: {
-        'appengine.googleapis.com/instance_name': process.env.GAE_INSTANCE,
-      },
-      resource: {
-        type: 'gae_app',
-        labels: {
-          module_id: process.env.GAE_SERVICE,
-          version_id: process.env.GAE_VERSION,
-        },
-      },
-      severity: 500, // Error.
-    };
-
     if (!debug) {
       res.sendStatus(statusCodes.ACCEPTED);
     }
 
     return unminify(stack, version)
       .then(stack => {
-        if (stack.length) {
-          event.message = event.message + `\n${stack.join('\n')}`;
+        const reqUrl =
+          req.method === 'POST'
+            ? `${req.originalUrl}?${queryString}`
+            : req.originalUrl;
+        const normalizedMessage = /^[A-Z][a-z]+: /.test(message)
+          ? message
+          : `Error: ${message}`;
+        const event = {
+          serviceContext: {
+            service: logTarget.serviceName,
+            version: logTarget.versionId,
+          },
+          message: [normalizedMessage, ...stack].join('\n'),
```

Comment:
Nit: Use concat. Spread iteration protocol is very slow.

Reply:
Done